testsuite/MDAnalysisTests/analysis/test_hydrogenbonds_analysis.py

```diff
@@ -87,15 +88,126 @@ def test_count_by_type(self, h):
+        most_common_hbond_ids = [12, 14, 9]
+        assert_equal(unique_hbonds[0,:3], most_common_hbond_ids)
+
         # count_by_ids() returns raw counts
         # convert to fraction of time that bond was observed
         counts = unique_hbonds[:, 3] / len(h.timesteps)
 
+        assert_equal(counts, ref_counts)
+
+
+
+class TestHydrogenBondAnalysisMock(object):
+
+    kwargs = {
+        'donors_sel': 'name O',
+        'hydrogens_sel': 'name H1 H2',
+        'acceptors_sel': 'name O',
+        'd_h_cutoff': 1.2,
+        'd_a_cutoff': 3.0,
+        'd_h_a_angle_cutoff': 120.0
+    }
+
+    @staticmethod
+    @pytest.fixture(scope='class')
+    def universe():
+        # create two water molecules
+        """
+                       H4
+                        \
+            O1-H2 .... O2-H3
+           /
+          H1
+        """
+        n_residues = 2
+        u = MDAnalysis.Universe.empty(
+            n_atoms=n_residues*3,
+            n_residues=n_residues,
+            atom_resindex=np.repeat(range(n_residues), 3),
+            residue_segindex=[0] * n_residues,
+            trajectory=True,  # necessary for adding coordinates
+            )
+
+        u.add_TopologyAttr('name', ['O', 'H1', 'H2'] * n_residues)
+        u.add_TopologyAttr('type', ['O', 'H', 'H'] * n_residues)
+        u.add_TopologyAttr('resname', ['SOL'] * n_residues)
+        u.add_TopologyAttr('resid', list(range(1, n_residues + 1)))
+        u.add_TopologyAttr('id', list(range(1, (n_residues * 3) + 1)))
+
+        # Atomic coordinates with a single hydrogen bond between O1-H2---O2
+        pos1 = np.array([[0, 0, 0],             # O1
+                        [-0.249, -0.968, 0],    # H1
+                        [1, 0, 0],              # H2
+                        [2.5, 0, 0],            # O2
+                        [3., 0, 0],             # H3
+                        [2.250, 0.968, 0]       # H4
+                        ])
+
+        # Atomic coordinates with no hydrogen bonds
+        pos2 = np.array([[0, 0, 0],             # O1
+                         [-0.249, -0.968, 0],   # H1
+                         [1, 0, 0],             # H2
+                         [4.5, 0, 0],           # O2
+                         [5., 0, 0],            # H3
+                         [4.250, 0.968, 0]      # H4
+                         ])
+
+        coordinates = np.empty((3,  # number of frames
+                                u.atoms.n_atoms,
+                                3))
+        coordinates[0] = pos1
+        coordinates[1] = pos2
+        coordinates[2] = pos1
+        u.load_new(coordinates, order='fac')
+
+        return u
+
+    def test_no_bond_info_exception(self, universe):
+
+        kwargs = {
+            'donors_sel': None,
+            'hydrogens_sel': None,
+            'acceptors_sel': None,
+            'd_h_cutoff': 1.2,
+            'd_a_cutoff': 3.0,
+            'd_h_a_angle_cutoff': 120.0
+        }
+
+        with pytest.raises(NoDataError):
```

Comment:
Could you please test for the correct error message to appear? Sorry for nit-picking but I think your nice error message is the crucial part here. `with pytest.raises(NoDataError, match=<text or regex>):` should do the trick.

Reply:
Thanks @zemanj , I've added it as requested. However, I would advice against it. This would make the test rely on strings, which should not have such power. It is the Exception type here that is clearly defined. If you mean to say that we should inherit a more specific Exception type, then that would be more appropriate. Anyway, it is a minor thing! Thanks, Mat

Reply:
I think that in general you're right, one usually shouldn't test for an error message. In this particular case, I think it's worth doing so. The reason is that we're soon gonna have `NoDataError`s for missing `AtomGroup` properties where appropriate, and I suspect that the same might eventually apply to `Universe` as well. Since your error message contains a lot of useful information, I want to avoid that it might be changed into a `NoDataError` with a generic error message in the future.